src/vscode-dts/vscode.d.ts

```diff
@@ -16081,6 +16081,24 @@ declare module 'vscode' {
 		 * @returns A promise that resolves to the Debug Adapter Protocol breakpoint or `undefined`.
 		 */
 		getDebugProtocolBreakpoint(breakpoint: Breakpoint): Thenable<DebugProtocolBreakpoint | undefined>;
+
+		/**
+		 * Obtains information on a possible data breakpoint that could be set on an expression or variable.
+		 * This will fail if the corresponding capability `supportsDataBreakpoints` is not supported by this session.
+		 *
+		 * @param name The name of the variable's child to obtain data breakpoint information for. If `variablesReference` isn't specified, this can be an expression.
+		 * @param variablesReference Reference to the variable container if the data breakpoint is requested for a child of the container.
+		 */
+		getDataBreakpointInfo(name: string, variablesReference?: number): Thenable<DataBreakpointInfo | undefined>;
```

Comment:
I don't think this method or the one below it should be in the VS Code API. Both are trivial wrappers around DAP calls and could be made via `customRequest`. We try to avoid taking unnecessary knowledge about DAP into VS Code API.

Reply:
I fully understand your point and agree that these methods are more like convenience wrappers around `customRequest` calls. The downside, however, is that with custom request calls we lack access to the session's capabilities, which means we cannot perform sanity checks on the client side (like RawDebugSession does) and have to rely on the debug adapter to send us proper errors. Would exposing these capabilities also be considered exposing unnecessary internal knowledge? I believe it could be useful in other scenarios like UI enablement as well.